playground/AzureServiceBus/ServiceBus.AppHost/Program.cs

```diff
@@ -0,0 +1,60 @@
+//using Aspire.Hosting.Azure.ServiceBus.ApplicationModel;
+
+var builder = DistributedApplication.CreateBuilder(args);
+
+var serviceBus = builder.AddAzureServiceBus("sbemulator")
+    .RunAsEmulator() // Comment to deploy and use the Azure cloud
+    ;
+
+serviceBus
+    .AddQueue("myQueue", queue =>
```

Comment:
@eerhardt In this case we would need to return an object that combines behavior of `IResourceBuilder<AzureServiceBusResource>` with what you are describing so that we could either modify the specified queue or start to specify a different queue or topic?

Reply:
(full disclosure, I'm not a super fan of fluent APIs)

If you want to start to specify a different queue, you would start a new code line.

```C#
var serviceBus = builder.AddAzureServiceBus("sbemulator");

var queue1 = serviceBus.AddQueue("queue1");
// set properties on queue1

var queue2 = serviceBus.AddQueue("queue2");
// set properties on queue2

Reply:
> Why doesn't .AddQueue return the Queue, and then you can modify it as you want?

This would be a breaking change. Is that something we'd want? But we could then reference the resources when building the model, or do `topic.AddSubscription("sub.1")`

Reply:
> This would be a breaking change. Is that something we'd want?

Any way we could make it without a breaking change? A new method name? `AddServiceBusQueue`? 

cc @mitchdenny @davidfowl - thoughts on using callbacks in the API this way?

Reply:
I had a chat with @DamianEdwards offline. We came to the conclusion that the current API isn't ideal. Ideally our APIs would follow either:

1. `parent.AddChild('name')` - returns the Child type (in this case a Queue)
or
2. `parent.WithChild('name', Action<Child> callback)` - returns the Parent type

The general rule - if you are returning a different type, use `Add`. If you are returning the same chained type, use `With`.

Since we already shipped `serviceBus.AddQueue('name')` returning the ServiceBus type, we could go with option 2 here.

The only drawback to 2 is that if you want to hold onto the `Child` in a variable, it gets messy. But the scenarios for needing a Queue/Topic/Subscription and using it elsewhere are limited. The only one I can think of is to get the name of the Queue and pass it into an environment variable to your app (or some other resource).